modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxManager.java

```diff
@@ -124,4 +128,20 @@ CompletableFuture<Void> cleanup(
      */
     @TestOnly
     int finished();
+
+    /**
+     * Updates the lower bound (exclusive) of the timestamp to start new read-only transactions.
+     *
+     * <p>All new read-only transactions will have to start strictly greater this lower bound.
+     *
+     * @param lowerBound New lower bound, {@code null} if there is no lower bound.
+     */
+    void updateLowerBoundToStartNewReadOnlyTransaction(@Nullable HybridTimestamp lowerBound);
```

Comment:
How about a different name, like `forbidReadOnlyTransactionsNotLaterThan()`? Just a stylistic matter.

Reply:
Maybe call it "updateLowWatermark" so that there is less guesswork, what do you think?